src/clients/python/experimental_api_v2/examples/simple_grpc_v2_sequence_infer_client.py

```diff
@@ -0,0 +1,192 @@
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from functools import partial
+import argparse
+import numpy as np
+import time
+import sys
+import queue
+
+import tritongrpcclient.core as grpcclient
+from tritongrpcclient.utils import InferenceServerException
+from tritongrpcclient.utils import get_stream_response_processor_pool
+
+FLAGS = None
+
+
+class UserData:
+
+    def __init__(self):
+        self._completed_requests = queue.Queue()
+
+
+# Callback function used for InferSequenceMetadata. Note the last
+# three parameters are reserved for InferResult, InferenceServerException
+# and the sequence ID for the response.
+def completion_callback(user_data, result, error, sequence_id):
+    user_data._completed_requests.put((result, error, sequence_id))
+
+
+def async_send(triton_client, response_pool, values, batch_size, sequence_id,
+               user_data, model_name, model_version):
+    # Prepare the sequence metadata object
+    sequence_metadata = grpcclient.InferSequenceMetadata(
+        sequence_id, partial(completion_callback, user_data))
+
+    # Add requests in the sequence for each value
+    count = 1
+    for value in values:
+        # Create the tensor for INPUT
+        value_data = np.full(shape=[batch_size, 1],
+                             fill_value=value,
+                             dtype=np.int32)
+        inputs = []
+        inputs.append(grpcclient.InferInput('INPUT'))
+        # Initialize the data
+        inputs[0].set_data_from_numpy(value_data)
+        outputs = []
+        outputs.append(grpcclient.InferOutput('OUTPUT'))
+        sequence_metadata.add_request(inputs,
```

Comment:
We don't want to have to put requests into a sequence before sending them. The 'sequence' object manages the lifetime of the GRPC stream (is stream is being used) but not the lifetime of requests. So async_sequence_infer should take same arguments as async_infer plus the sequence (required).